openmmtools/multistate/sams.py

```diff
@@ -171,10 +172,14 @@ def __init__(self,
                  log_target_probabilities=None,
                  state_update_scheme='global-jump',
                  locality=5,
-                 update_stages='two-stage',
-                 flatness_criteria='logZ-flatness',
-                 flatness_threshold=0.2,
+                 weight_update=True,
                  weight_update_method='rao-blackwellized',
+                 update_stages='two-stage',
+                 histogram_flatness = None,
+                 minimum_visits = None,
+                 minimum_round_trips = None,
```

Comment:
I think the errors are due to a bunch of tests that fail because right now by default no condition for switching stage is set. Maybe we can change the defaults to use our "best practice" criteria?

Reply:
You can ignore the MacOS failures btw. Only linux is working right now.

Reply:
Ok. Could I set the histogram flatness as the default?

Reply:
Yeah, that sounds great.

Reply:
When running pytest, I am getting the following error (test_sampling.py:1060: AssertionError) for all samplers:
```
E                 Full diff:
E                 {'_adapt_target_probabilities': False,
E                 '_beta_factor': 0.8,
E                 '_downhill': False,
E                 '_gamma0': 1.0,
E                 '_have_displayed_citations_before': False,
E                 '_histogram_flatness': 0.3,
E                 '_iteration': 3,
E                 -  '_last_err_free_energy': inf,
E                 ?                           ^^^
E                 +  '_last_err_free_energy': masked,
E                 ?                           ^^^^^^
```

Reply:
We're actually still using `nose` with openmmtools. We've not converted all the tests to `pytest` yet. Not sure if this is going to help, but it's worth the shot.